src/main/resources/Diagram/DiagramEditSheet.xml

```diff
@@ -715,6 +715,24 @@ define('diagramStore', ['jquery', 'xwiki-meta', 'draw.io', 'xwiki-events-bridge'
     oldImportFiles.apply(this, newArgs);
   }
 
+  //
+  // Change the URL generated by export for containing the actual host
+  //
+  var oldCreateLink = EditorUi.prototype.createLink;
+    EditorUi.prototype.createLink = function() {
+    let rawURL = oldCreateLink.apply(this, arguments);
+    var docURL = new XWiki.Document(new XWiki.DocumentReference('', 'Diagram', 'DiagramViewSheet')).getURL();
+    let actualURL = rawURL.replace('https://www.draw.io/', window.location.protocol + window.location.host + docURL);
+    let index = actualURL.indexOf('#R');
+    if (index &gt; -1) {
+      // Add a parameter in query to be checked in DiagramViewSheet
+      actualURL = [actualURL.slice(0, index), (actualURL.indexOf('?') &gt; -1) ? '&amp;export=url' : '?export=url',
```

Comment:
I would have used "source" instead of "export" because the diagram was already exported as URL and we're now viewing it and the viewer needs to know where to read the diagram from (from the content of the current page, from the URL, etc.). So for me this parameter tells the viewer that the diagram source is on the URL.